src/main/java/com/denizenscript/denizencore/scripts/commands/core/MongoCommand.java

```diff
@@ -0,0 +1,521 @@
+                        return;
+                    }
+                    DenizenCore.schedule(new AsyncSchedulable(new OneTimeSchedulable(() -> {
+                        MongoClient con = null;
+                        String conStr = uri.getValue();
+                        if (!conStr.startsWith("mongodb://") && !conStr.startsWith("mongodb+srv://")) {
+                            conStr = "mongodb://" + conStr;
+                        }
+                        if (CoreConfiguration.debugVerbose) {
+                            Debug.echoDebug(scriptEntry, "Connecting to Mongo server...");
+                        }
+                        MongoDatabase db = null;
+                        MongoCollection<Document> col = null;
+                        try {
+                            con = MongoClients.create(conStr);
+                            db = con.getDatabase(database);
+                            col = db.getCollection(collection);
+                        }
+                        catch (final Exception e) {
+                            DenizenCore.runOnMainThread(() -> {
+                                Debug.echoError(scriptEntry, "Mongo Exception: " + e.getMessage());
+                                scriptEntry.setFinished(true);
+                                if (CoreConfiguration.debugVerbose) {
+                                    Debug.echoError(scriptEntry, e);
+                                }
+                            });
+                        }
+                        if (CoreConfiguration.debugVerbose) {
+                            Debug.echoDebug(scriptEntry, "Connection did not error.");
+                        }
+                        final MongoClient conn = con;
+                        final MongoDatabase connDB = db;
+                        final MongoCollection<Document> coll = col;
+                        if (con != null) {
+                            DenizenCore.runOnMainThread(() -> {
+                                connections.put(connectionId, conn);
+                                databases.put(connectionId, connDB);
+                                collections.put(connectionId, coll);
+                                Debug.echoDebug(scriptEntry, "Successfully connected to Mongo server.");
+                                scriptEntry.setFinished(true);
+                            });
+                        }
+                        else {
+                            DenizenCore.runOnMainThread(() -> {
+                                Debug.echoDebug(scriptEntry, "Connecting errored!");
+                                scriptEntry.setFinished(true);
+                            });
+                        }
+                    }, 0)));
+                }
+                // If disconnect is true, it means it is present.
+                else if (disconnect) {
+                    MongoClient con = connections.get(connectionId);
+                    if (con == null) {

```

Comment:
Since this check is duplicated for everything other than the `connect` option, can just `return;` from the `connect` block and then handle the `con` grab outside the `if`